4_predict/src/plot_diagnostics.R

```diff
@@ -0,0 +1,559 @@
+  #' @param out_dir output directory
+  #'
+  #' @return filepath to the resulting plot
+  
+  fileout <- file.path(out_dir, paste0('compare_models_', model_name, '_', pred_var, '_', perf_metric, '_CV.png'))
+  
+  #CV performances dataframe
+  plt_df <- data.frame(perf = c(tune::show_best(mod$grid_params, n = 1, metric = perf_metric)$mean,
+                                get_perf_metric(mod$best_fit$.metrics[[1]], perf_metric = perf_metric)
+                                ),
+                       sd = c(tune::show_best(mod$grid_params, n = 1, metric = perf_metric)$std_err,
+                              NA),
+                       Dataset = c('Val', 'Test'),
+                       grp = c("RF-Static","RF-Static"))
+  
+  p1 <- ggplot(data = plt_df, aes(x = grp, y = perf, fill = Dataset)) +
+    geom_bar(stat="identity", position=position_dodge(), width = 0.6) +
+    theme_bw() +
+    scale_fill_brewer(palette="Paired") +
+    geom_errorbar(aes(ymin = perf - 2*sd, ymax = perf + 2*sd), width = .2,
+                  position = position_dodge(0.6)) +
+    xlab('') +
+    ylab(perf_metric) + 
+    scale_x_discrete(limits=c("RF-Static")) +
+    theme(axis.title.y = element_text(size = 14),
+          axis.text.x = element_text(size = 14)) +
+    ggtitle(paste0(model_name, ', ', pred_var))
+  
+  ggsave(filename = fileout, plot = p1, device = 'png')
+  
+  return(fileout)
+}
+
+get_perf_metric <- function(model_fit, perf_metric){
+  #' @description returns the performance metric for a fitted model
+  #'
+  #' @param model_fit fitted model
+  #' @param perf_metric performance metric name
+  #'
+  #' @return performance metric value
+  
+  model_fit$.estimate[model_fit$.metric == perf_metric]
+}
+
+plot_barplot <- function(attr_data, file_path,
+                         model_name, plot_month_names = FALSE,
+                         panel = FALSE, label_sequence = NULL){
+  #' 
+  #' @description Creates a barplot for each of the columns in attr_data
+  #'
+  #' @param attr_data data frame for which the first column is the x-axis of the 
+  #' barplot and the remaining columns are y-axes
+  #' @param file_path a character string that indicates the location of the saved plot
+  #' @param model_name character string describing the model. Will be added 
+  #' to the end of the filename before the file extension, and also be the plot title.
+  #' @param plot_month_names if TRUE, then the x-axis will be converted from
+  #' numeric to month.abb month names
+  #' @param panel if TRUE, columns 2 and 3 of attr_data will be plotted in a
+  #' panel plot.
+  #' @param label_sequence the indices of labels to plot on the x-axis. NULL
+  #' plots all labels, which can get crowded for some plots.
+  #'
+  #' @return Returns the path to png files containing barplots of each attribute
+  
+  plot_names <- vector('character', length = 0L)
+  plt_lst <- list()
+  
+  # For each column, create a barplot
+  for(i in 2:dim(attr_data)[2]){
```

Comment:
I think I'll leave this function as-is for now because I'd need to add another argument to specify the x-axis column name to use `seq_along` for the y-axes. I did implement it for the nhdv2 attr plotting, though.